Build failure on non-x86*: "comparison is always true due to limited range of data type [-Werror=type-limits]"

```
In file included from /builds/PureTryOut/aports/testing/valhalla/src/valhalla-3.1.0_git20210304/valhalla/baldr/graphid.h:11,
                 from /builds/PureTryOut/aports/testing/valhalla/src/valhalla-3.1.0_git20210304/valhalla/baldr/directededge.h:6,
                 from /builds/PureTryOut/aports/testing/valhalla/src/valhalla-3.1.0_git20210304/valhalla/sif/pedestriancost.h:5,
                 from /builds/PureTryOut/aports/testing/valhalla/src/valhalla-3.1.0_git20210304/src/sif/pedestriancost.cc:1:
/builds/PureTryOut/aports/testing/valhalla/src/valhalla-3.1.0_git20210304/valhalla/baldr/json.h: In member function 'std::ostream& valhalla::baldr::json::OstreamVisitor::operator()(const string&) const':
/builds/PureTryOut/aports/testing/valhalla/src/valhalla-3.1.0_git20210304/valhalla/baldr/json.h:97:17: error: comparison is always true due to limited range of data type [-Werror=type-limits]
   97 |           if (c >= 0 && c < 32) {
      |               ~~^~~~
```

```
/builds/PureTryOut/aports/testing/valhalla/src/valhalla-3.1.0_git20210304/third_party/rapidjson/include/rapidjson/writer.h:430:67: error: comparison is always true due to limited range of data type [-Werror=type-limits]
  430 |             else if ((sizeof(Ch) == 1 || static_cast<unsigned>(c) < 256) && RAPIDJSON_UNLIKELY(escape[static_cast<unsigned char>(c)]))  {
      |                                          ~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~
```

This only seems to happen on non-x86 and x86_64 arches like aarch64 and ppc64le.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -2,6 +2,7 @@
 * **Removed**
 * **Bug Fix**
    * FIXED: valhalla_run_route was missing config logic.[#3824](https://github.com/valhalla/valhalla/pull/3824)
+   * FIXED: Handle hexlifying strings with unsigned chars [#3842](https://github.com/valhalla/valhalla/pull/3842)
 * **Enhancement**
    * ADDED: Add time info to sources_to_targets [#3795](https://github.com/valhalla/valhalla/pull/3795)
    * ADDED: "available_actions" to the /status response [#3836](https://github.com/valhalla/valhalla/pull/3836)

--- a/valhalla/baldr/json.h
+++ b/valhalla/baldr/json.h
@@ -2,6 +2,7 @@
 #define VALHALLA_BALDR_JSON_H_
 
 #include <boost/variant.hpp>
+#include <cctype>
 #include <cinttypes>
 #include <cmath>
 #include <cstddef>
@@ -110,7 +111,7 @@ public:
           ostream_ << "\\t";
           break;
         default:
-          if (c >= 0 && c < 32) {
+          if (iscntrl(c)) {
             // format changes for json hex
             ostream_.setf(std::ios::hex, std::ios::basefield);
             ostream_.setf(std::ios::uppercase);

